documentation/specs/TextField.md

```diff
@@ -0,0 +1,307 @@
+# `TextField` Component Specification
+
+## Overview
+
+The `TextField` component allows users to input text on a single line and provides the essentials to be used as a form control element. An Easy UI password input variation can be had by setting `variant="protected"`.
+
+### Use Cases
+
+- Use this component when building forms that require `text`, `email`, `password`, and `tel` inputs.
+- Can be used as the basis for a `search` input.
+
+### Features
+
+- The net effect of setting `variant` to `protected` is to get a PasswordInput component with an appropriate visibility icon.
```

Comment:
i suppose we could use the input `type` prop to accomplish the same thing instead of having a `protected` variation, but i feel that since our password inputs have the clickable visibility icon, it makes a bit more sense to me to capture that info into a separate prop, but open to ideas

Reply:
i don't think i have a preference. your reasoning seems sound to me.

out of curiosity, would a `variant` ever be `protected` when the `type` is not `password`?

Reply:
good question. i checked with design and there are no plans as of yet to want to support a password input that deviates from whats currently in figma, so i removed the `variant` property